src/telegraf.ts

```diff
@@ -185,40 +257,44 @@ export class Telegraf<C extends Context = Context> extends Composer<C> {
     this.polling?.stop()
   }
 
-  private handleUpdates(updates: readonly tt.Update[]) {
-    if (!Array.isArray(updates)) {
-      throw new TypeError(util.format('Updates must be an array, got', updates))
-    }
-    // prettier-ignore
-    const processAll = Promise.all(updates.map((update) => this.handleUpdate(update)))
-    if (this.options.handlerTimeout === Infinity) {
-      return processAll
-    }
-    return Promise.race([processAll, wait(this.options.handlerTimeout)])
+  async handleUpdate(update: tt.Update, webhookResponse?: http.ServerResponse) {
+    await this.handleUpdates([update], webhookResponse)
```

Comment:
Does this mean that each update delivered via webhook will spawn a new timer?

Reply:
No. There is only ever one timer at most per bot, no matter how many updates are processed.

Instead, what this implies is that a timer will time out only one update in the same iteration, never batches of updates (unless they arrive in the same millisecond, that is).

Reply:
Then, does this mean that if 5 updates arrive, 5th one has 5 times as long timeout?

Reply:
No. Let's assume the middleware does not ever complete for this example:

Polling:
- five updates arrive
- one timer is set
- all updates will be purged together
- no timer is running

Webhooks:
- first update arrives
- one timer is set
- other updates arrive (timer is not touched)
- first update will be purged
- timer is restarted for the remaining time until the second update times out
- second update is purged
- timer is restarted for the remaining time until the third update times out
- etc
- last update is purged
- no timer is running